src/bonds.cpp

```diff
@@ -133,6 +136,18 @@ void FENEBond::setEnergyFunction(const ParticleVector& particles) {
         }
         return -half_force_constant * max_squared_distance * std::log(1.0 - squared_distance / max_squared_distance);
     };
+    forceFunc = [&](Geometry::DistanceFunction distance) -> std::vector<IndexAndForce> {
+      const Point ba = distance(particles[indices[0]].pos, particles[indices[1]].pos); // b->a
+      const auto squared_distance = ba.squaredNorm();
+      if (squared_distance >= max_squared_distance) {
+          throw std::runtime_error("Fene potential: Force undefined for distances greater than rmax.");
+      };
+      const auto magnitude =
+          -2.0 * half_force_constant * ba.norm() / (1.0 - squared_distance/max_squared_distance);
+      Point force0 = magnitude * ba.normalized();
+      Point force1 = -force0;
+      return {{indices[0], force0}, {indices[1], force1}};
+    };
 }
 
 FENEWCABond::FENEWCABond(double k, double rmax, double epsilon, double sigma, const std::vector<int>& indices)
```

Comment:
use descriptive argument names

Reply:
Related to https://github.com/mlund/faunus/pull/368#discussion_r638527561.
Fixed with f1feccc.